lib/task_helpers/development/replication.rb

```diff
@@ -0,0 +1,126 @@
+require 'awesome_spawn'
+
+module TaskHelpers
+  module Development
+    class Replication
+      REMOTES = [1, 2]
+      GLOBAL  = 99
+      GUID_FILE   = Rails.root.join("GUID")
+      BACKUP_GUID = Rails.root.join("GUID.backup")
+
+      PG_USER   = "root"
+      PG_PASS   = "smartvm"
+      PG_HOST   = "localhost"
+      PG_PORT   = "5432"
+      DB_PREFIX = "development_replication"
+
+      class << self
+        def backup
+          if File.exist?(guid_file)
+            FileUtils.rm_f(backup_guid)
+            FileUtils.mv(guid_file, backup_guid)
+          end
+        end
+
+        def restore
+          if File.exist?(backup_guid)
+            FileUtils.rm_f(guid_file)
+            FileUtils.mv(backup_guid, guid_file)
+          end
+        end
+
+        def setup
+          REMOTES.each {|r| setup_remote(r) }
+          setup_global(GLOBAL)
+
+          # TODO: We have the technology to watch for this and report when it's all good or bad
+          puts "Local replication is setup... try checking for users in all regions: psql -U root development_replication_99 -c \"select id from users;\""
+        end
+
+        def teardown
+          REMOTES.each do |r|
+            teardown_global_subscription_for_region(r)
+            teardown_remote_publication(r)
+          end
+        end
+
+        def setup_remote(region)
+          setup_one_region(region)
+          setup_remote_region(region)
+        end
+
+        def setup_global(region)
+          setup_one_region(region)
+          setup_global_region(region)
+        end
```

Comment:
With the other naming suggestions, I think this reads a little better:

```suggestion
        def setup_remote(region)
          create_region(region)
          configure_remote_region(region)
        end

        def setup_global(region)
          create_region(region)
          configure_global_region(region)
        end
```